metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/configs/HiveConnectorClientConfig.java

```diff
@@ -128,30 +133,34 @@ public DataSourceTransactionManager hiveTxManager(
     }
 
     /**
-     * hive metadata JDBC template.
+     * hive metadata read JDBC template. Query timeout is set to control long running read queries.
      *
+     * @param connectorContext connector config.
      * @param hiveDataSource hive data source
      * @return hive JDBC Template
      */
     @Bean
-    public JdbcTemplate hiveJdbcTemplate(
+    public JdbcTemplate hiveReadJdbcTemplate(
+        final ConnectorContext connectorContext,
         @Qualifier("hiveDataSource") final DataSource hiveDataSource) {
-        return new JdbcTemplate(hiveDataSource);
+        final JdbcTemplate result = new JdbcTemplate(hiveDataSource);
+        result.setQueryTimeout(getDataStoreReadTimeout(connectorContext) / 1000);
```

Comment:
Why not set as second to avoid conversion there?

Reply:
I am reusing the jdo parameters that already existed.